Source/Core/DolphinQt2/Config/Graphics/GraphicsWidget.cpp

```diff
@@ -0,0 +1,23 @@
+// Copyright 2017 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include "DolphinQt2/Config/Graphics/GraphicsWidget.h"
+
+#include <QEvent>
+#include <QFormLayout>
+#include <QGroupBox>
+#include <QLabel>
+#include <QVBoxLayout>
+
+#include "DolphinQt2/Config/Graphics/GraphicsWindow.h"
+
+GraphicsWidget::GraphicsWidget(GraphicsWindow* parent)
+{
+  parent->RegisterWidget(this);
```

Comment:
Having a child object reference the parent object like this increases coupling and is a bad pattern. Since it's being done to expose information during construction, one way to fix this would be to move the information out of the constructor. You could do something like:

```c++
class MyWidget
{
  GetDescriptions(std::function<void(QWidget*, const char*)> add_description)
  {
    add_description(child_widget, description);
    // ...
  }
};

class ParentWidget
{
  ParentWidget()
  {
    m_my_widget = new MyWidget();
    MyWidget->GetDescriptions(
        [=](QWidget* widget, const char* description) { OnDescriptionAdded(widget, description); });
  }
};
```

instead of the current

```c++
class MyWidget
{
  MyWidget(ParentWidget* parent)
  {
    parent->RegisterWidget(this);
    emit DescriptionAdded(child_widget, description);
    // ...
  }

signals:
  void DescriptionAdded(QWidget* widget, const char* description);
};

class ParentWidget
{
  ParentWidget() { m_my_widget = new MyWidget(this); }
  void RegisterWidget(MyWidget* my_widget)
  {
    connect(my_widget, &MyWidget::DescriptionAdded, this, &ParentWidget::OnDescriptionAdded);
  }
};
```

(which also has an additional benefits of being more direct and explicit).